app/src/org/commcare/tasks/ProcessAndSendTask.java

```diff
@@ -347,6 +343,10 @@ private void sendForms(FormRecord[] records) throws TaskCancelledException {
                             break;
                         }
                         continue;
+                    } catch (IOException e) {

```

Comment:
Hey,

I don't think this is quite safe enough as a processing path to be comfortable. There's a lot of IO that gets handled in this method, and it would be very easy for someone to introduce and I/O issue between lines 298 and 356 that they don't realize is actually unhandled. 

I'm gonna change this around just a bit and propose a different pattern